### ISSUE TITLE
Run yolov8 with gpu on pc more easy

### ISSUE THREAD

**Issue by kyoko9000** [ultralytics/ultralytics#1857]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Now if i need to run yolov8 with gpu on pc i have to install library manual. But so many people try but it hard to do. So i need to run with gpu on pc i just choose gpu some where on the code and yolov8 can auto download and install pack of gpu. If intergrate gpu on normal yolo pack, it will be heavy no need, just some one got gpu. 

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
GPUs should work easily with ultralytics. And no extra installation steps for GPU are needed. Once you've installed and verified your Pytorch CUDA (can be verified using `torch.cuda.is_available()`), you should be able to simply pass `device=0` for sinlgle GPU operations.

```
model = YOLO("yolov8n.pt")
model.train(device='0')
```

**Comment by kyoko9000** [comment]
It mean i no need to install any thing. If i want to detect webcam with my gpu what should i do?

**Comment by Manueljohnson063** [comment]
while cap.isOpened():
    start_time = time.time()
    success, frame = cap.read()
    stop_time = time.time()
    total_time = stop_time - start_time
    print("Total time taken:", total_time, "seconds")

    if success:
        gpu_frame = cv2.cuda_GpuMat(frame)
        #gpu_bgr_frame = cv2_cuda.cvtColor(gpu_frame, cv2.COLOR_BGR2BGRA)
        **results = model(gpu_frame, stream=False, device=0, conf=0.1, iou=0.1, verbose=False)[0]**

        masks = results.masks 
Is this possible

**Comment by pderrenger** [comment]
@Manueljohnson063 your code snippet seems to be attempting to use YOLOv8 for real-time object detection on a webcam feed with a GPU. However, YOLOv8, as of now, doesn't directly support CUDA operations. The model is run on CPU with an optional parameter to set the device to either 0 for the first GPU or "cpu" for CPU. Also, the YOLOv8 model directly accepts OpenCV frames, so there is no need to explicitly convert the frame to a GPU mat. You can simply use the predict method of the YOLOv8 model passing in the frame from the webcam and then use the results obtained for further processing.
